src/Compiler/Microsoft.AspNetCore.Razor.Language/src/RazorCodeDocumentExtensions.cs

```diff
@@ -121,7 +122,7 @@ public static void SetSyntaxTree(this RazorCodeDocument document, RazorSyntaxTre
         document.Items[typeof(RazorSyntaxTree)] = syntaxTree;
     }
 
-    public static IReadOnlyList<RazorSyntaxTree> GetImportSyntaxTrees(this RazorCodeDocument document)
+    public static ImmutableArray<RazorSyntaxTree>? GetImportSyntaxTrees(this RazorCodeDocument document)
```

Comment:
I was trying to match the original semantics to make it easier to update the callsites, since it's possible for `default` to be passed to `ImmutableArray<RazorSyntaxTree>`. Each callsite currently checks for `null`, though they could be changed to check `IsDefault`. I'll take a look.

Reply:
I've made this change, but to be honest, I don't love it. :smile: